### ISSUE TITLE
Add Crouzeix-Raviart basis

### ISSUE THREAD

**Issue by lindsayad** [libMesh/libmesh#3318]
Crouzeix-Raviart sounds like an interesting candidate for solving  incompressible Navier-Stokes problems as described briefly [here](https://fenicsproject.org/pub/book/book/fenics-book-2011-06-14.pdf) and in more detail elsewhere

**Comment by lindsayad** [comment]
In my mind these styles of discretizations seem somewhat related to a staggered grid finite volume scheme

**Comment by lindsayad** [comment]
More theory around CR [here](https://www.math.tamu.edu/~guermond/M661_FALL_2015/chap24.pdf). Thanks to @jwpeterson for these great references

**Comment by lindsayad** [comment]
[This article](https://link.springer.com/content/pdf/10.1007/s11464-008-0034-9.pdf) by Rui and Bi discusses a finite volume element method. I've been seeing quite a bit of dual mesh language in which a second set of elements/cells is built around the barycentric centers of cell faces. So if these methods require a dual mesh in order to function, it makes just as much sense to go straight ahead with a staggered-grid pure finite volume method IMO

**Comment by lindsayad** [comment]
Staggered Semi-Implicit Hybrid Finite Volume/Finite Element Schemes for Turbulent and Non-Newtonian Flows [by Busto](https://mdpi-res.com/d_attachment/mathematics/mathematics-09-02972/article_deploy/mathematics-09-02972-v2.pdf?version=1637806891) looks like another interesting piece of literature although unrelated to CR or RT. It talks in-depth about construction of dual meshes for both simplex and quad meshes

**Comment by GiudGiud** [comment]
I need to read up on this. I'd be down for a summary table of all the discretizations we consider, what kind of interpolation they end up requiring (if any), the storage requirements, whether it s been deployed successfully elsewhere (less risk, less reward imo since less pushing the state of the art), etc all that kind of metadata which would help us make a decision

**Comment by lindsayad** [comment]
I forget whether these bases have the element constants in their test spaces...

**Comment by jwpeterson** [comment]
> I forget whether these bases have the element constants in their test spaces...

By "element constants", do you mean the ability to exactly represent a piecewise-discontinuous-constant function on the elements? Since the CR functions are continuous at the mid-edge nodes, I don't think this is possible in the way that it would be for "true" discontinuous bases like `MONOMIAL`.

**Comment by lindsayad** [comment]
Yes after careful thought about what I meant ... that is what I meant haha

**Comment by jwpeterson** [comment]
OK, well it's an interesting question. I guess we don't need this property in order to prove convergence, etc. of the element so I'm curious why you are curious about it.

**Comment by roystgnr** [comment]
IIRC there are a lot of proofs (of convergence, global conservation, etc) that do rely on having u=c in the trial space, but all you need is the per-connected-domain-subset constant, not per-element constant.

**Comment by lindsayad** [comment]
My motivation for the question was this: my understanding based on Long Chen's writing [here](https://www.math.uci.edu/~chenlong/226/FEMStokes.pdf) is that in order to have element-wise mass conservation, you need to have the piecewise constants in the test space for pressure.

We are writing up a paper on the finite volume method and there has been a fair amount of discussion on the local conservation (or lack thereof) of Taylor-Hood elements, what I think is often colloquially referred to simply as CFEM. It is very common for people to say that CFEM is not locally conservative. However, this nice article by [White](https://link.springer.com/content/pdf/10.1007/s10596-021-10044-3.pdf) is one in a line of articles that try to debunk that myth, contenting that CFEM is indeed locally conservative ... it's just not on a per-element basis, but instead on a nodal-patch basis (e.g. following the shape of the test functions).

**Comment by jwpeterson** [comment]
> you need to have the piecewise constants in the test space for pressure.

OK, yeah, from what I've seen the CR element always uses the non-conforming basis functions for the velocity components, and constant, monomial pressures. I assumed this was just because of LBB-stability requirements, but it would not surprise me if you also needed it to guarantee element-local conservation properties.

**Comment by lindsayad** [comment]
In the vein of conservation discussion is Hughes's work [The Continuous Galerkin Method is Locally Conservative](https://reader.elsevier.com/reader/sd/pii/S002199910096577X?token=8B8621FFE93486A19B76C1C270B90598BB2A5D2EEFD44A281EE78DF4FFE1C7A6AAA34A8668654602F55C113377BF47E0&originRegion=us-east-1&originCreation=20220725210600)
